tests/framework/extensions/clusters/clusters.go

```diff
@@ -405,6 +412,38 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace string, clustersConfig *Clus
 	return v1Cluster
 }
 
+func awsSystemConfigHelper() (rkeConfig []rkev1.RKESystemConfig) {
```

Comment:
I think this is pretty close to what I commented, just to add a bit more functionality to this, I think we could ask for roles and keys as args to this function. And make roles and keys const or enum too. That way we would see how this constructor works and what it requires from the outer scope, i.e. while reading `NewK3sRKE2ClusterConfig` for the first time.

But just like my previous comment, it really depends if we're going to use the same roles and keys in `configData` later on for the other providers.

Also, a small nit I don't think we need to have a `Helper` suffix for this, I think `awsSystemConfig` is pretty explanatory, or if you want to declare `newAwsSystemConfig`.

Reply:
my only push back on this is, I don't think it makes sense to have the roles/keys as an arg since this is specific to AWS. The only other out-of-tree use case we currently test is vsphere, which doesn't require these changes to the systemConfig.